### ISSUE TITLE
seg fault if output configuration fails

### ISSUE THREAD

**Issue by dsbrng25b** [fluent/fluent-bit#1195]
## Bug Report

**Describe the bug**
If you have a filter in your configuration and the configuration of an output fails it causes a segmentation fault. It's not a big problem because in this case you haven't a running configuration anyway but its not nice.

**To Reproduce**
```
fluent-bit -i systemd -F nest -o splunk
# or
fluent-bit -i systemd -F grep -o splunk
# or
fluent-bit -i systemd -F grep -o azure
# ...
```

**Expected behavior**
fluent-bit aborts normally without segmentation fault.

**Your Environment**

* Version used: v1.1.0 (881ca2a802dc7f40fd7fe5f6704211923ae72942)
* Operating System and version: Ubuntu 18.04.2 LTS Linux  4.15.0-45-generic #48-Ubuntu SMP Tue Jan 29 16:28:13 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by dominique-pfister** [comment]
#1227 fixes the seg fault

**Comment by edsiper** [comment]
@dominique-pfister thanks for work on this issue. 

Despite the fix is valid in #1227, the problem another root cause that was already addressed in:

- 7b60b49914ce64e9809b315ef673d5f59685f201
- 965b0bd3405473211b6cfa197565494bea12a149

Basically is to initialize the plugin instance with zeros (using calloc(2) ) plus avoiding to invoke the callback exit of a filter if NO context has been defined. 

thanks @dsbrng25b and @dominique-pfister for work on this issue.
